reference-implementation/lib/writable-stream.js

```diff
@@ -21,23 +21,31 @@ class WritableStream {
     this._writeRequests = [];
 
     // Write requests are removed from _writeRequests when write() is called on the underlying sink. This prevents
-    // them from being erroneously rejected on error. If a write() call is pending, the request is stored here.
-    this._pendingWriteRequest = undefined;
+    // them from being erroneously rejected on error. If a write() call is in-flight, the request is stored here.
+    this._inFlightWriteRequest = undefined;
 
     // The promise that was returned from writer.close(). Stored here because it may be fulfilled after the writer
     // has been detached.
-    this._pendingCloseRequest = undefined;
+    this._closeRequest = undefined;
+
+    // Close request is removed from _closeRequest when close() is called on the underlying sink. This prevents it
+    // from being erroneously rejected on error. If a close() call is in-flight, the request is stored here.
+    this._inFlightCloseRequest = undefined;
```

Comment:
> I guess I should try this after landing the patch.

+1 for getting this landed before changing anything else.

I think in practice all browser implementations are going to pack the internal slots somewhat. Chrome does, and if I understand correctly Firefox goes further and defines the object layout using native code. I think that the key things for the standard are finding the representation of the state that is easiest to understand and reason about, and explicitly mentioning things like "only one of [[closeRequest]] and [[inFlightWriteRequest]] is used at a time" to help other implementations optimise.